servers/src/main/java/tachyon/worker/block/BlockServiceHandler.java

```diff
@@ -0,0 +1,233 @@
+    } catch (IOException ioe) {
+      throw new TException(ioe);
+    }
+  }
+
+  /**
+   * Used to get user's temporary folder on under file system, and the path of the user's temporary
+   * folder will be returned.
+   *
+   * @param userId the id of the user requesting the ufs location
+   */
+  public String getUserUfsTempFolder(long userId) {
+    return mWorker.getUserUfsTmpFolder(userId);
+  }
+
+  /**
+   * Lock the file in Tachyon's space while the user is reading it, and the path of the block file
+   * locked will be returned, if the block file is not found, FileDoesNotExistException will be
+   * thrown.
+   *
+   * @param blockId the id of the block to be locked
+   * @param userId the id of the
+   * @throws FileDoesNotExistException if the block does not exist
+   */
+  public String lockBlock(long blockId, long userId) throws FileDoesNotExistException {
+    try {
+      long lockId = mWorker.lockBlock(userId, blockId);
+      return mWorker.readBlock(userId, blockId, lockId);
+    } catch (IOException ioe) {
+      throw new FileDoesNotExistException("Block " + blockId + " does not exist on this worker.");
+    }
+  }
+
+  /**
+   * Used to promote block on under storage layer to top storage layer when there are more than one
+   * storage layers in Tachyon's space. return true if the block is successfully promoted, false
+   * otherwise.
+   *
+   * @param blockId the id of the block to move to the top layer
+   * @throws TException if the block cannot be moved to the top layer
+   */
+  // TODO: This may be better as void
+  public boolean promoteBlock(long blockId) throws TException {
+    try {
+      // TODO: Make the top level configurable
+      mWorker.moveBlock(Users.MIGRATE_DATA_USER_ID, blockId, StorageLevelAlias.MEM.getValue());
+      return true;
+    } catch (IOException ioe) {
+      throw new TException(ioe);
+    }
+  }
+
+  /**
+   * Used to allocate location and space for a new coming block, worker will choose the appropriate
+   * storage directory which fits the initial block size by some allocation strategy, and the
+   * temporary file path of the block file will be returned. if there is no enough space on Tachyon
+   * storage OutOfSpaceException will be thrown, if the file is already being written by the user,
+   * FileAlreadyExistException will be thrown.
+   *
+   * @param userId the id of the client requesting the create
+   * @param blockId the id of the new block to create
+   * @param initialBytes the initial number of bytes to allocate for this block
+   * @throws OutOfSpaceException if the worker cannot provide enough space to fit the block
+   */
+  public String requestBlockLocation(long userId, long blockId, long initialBytes)
+      throws OutOfSpaceException {
+    try {
+      // NOTE: right now, we ask allocator to allocate new blocks in MEM tier
+      return mWorker.createBlock(userId, blockId, StorageLevelAlias.MEM.getValue(), initialBytes);
+    } catch (IOException ioe) {
```

Comment:
what about LOG.error this `ioe`, it's a bit hard to figure out why failing to allocate space since all errors are in `IOException` without LOG.error.